legate/core/store.py

```diff
@@ -1119,6 +1131,10 @@ def slice(self, dim: int, sl: slice) -> Store:
             )
 
         size = self.shape[dim]
+
+        if size == 0 and sl == slice(None):
```

Comment:
I think we can change this to `if size == 0 or sl == slice(None):`, as a slicing is no-op with either of them.

FYI, it looks like NumPy allows slice indices to be out-of-bounds, which would end up resulting an empty slice, whereas we raise an exception below. This is something we should fix after this PR is merged.

Reply:
makes sense.